### PR TITLE
Adds LLVMVer for version specification with ppLLVM and LLVMPretty Ident instance.

### PR DESCRIPTION
**Pull request by kquick**
This change adds an `LLVMVer` definition to allow specification of LLVM versions beyond the previous 3.5, 3.6, 3.7, and 3.8.  The `ppLLVM` function is updated to take this additional argument to determine the format of the LLVM output.

This also adds an `LLVMPretty` instance for `Ident`.

This is part of a series of pretty-printing updates, and more PR's will be forthcoming.

### CODE REVIEW

File: src/Text/LLVM/PP.hs [hunk 1]
```diff
@@ -34,6 +34,46 @@ import Prelude hiding ((<>))
...
+llvmV3_5 = 0
+llvmV3_6 = 1
+llvmV3_7 = 2
+llvmV3_8 = 3
+
+-- | This value should be updated when support is added for new LLVM versions;
+-- this is used for defaulting and otherwise reporting the maximum LLVM version
+-- known to be supported.
+llvmVlatest :: LLVMVer
+llvmVlatest = 16
```
Comment (RyanGlScott) [inline src/Text/LLVM/PP.hs:74]: Is this used anywhere?

File: src/Text/LLVM/PP.hs [hunk 2]
```diff
@@ -45,29 +85,43 @@ data Config = Config { cfgLoadImplicitType :: Bool
                        -- instruction is implied.
 
                      , cfgUseDILocation :: Bool
+
+                     , cfgVer :: LLVMVer
```
Comment (RyanGlScott) [inline src/Text/LLVM/PP.hs:89]: I'd be willing to bet that all of the fields above this one (`cfgLoadImplicitType`, `cfgGEPImplicitType`, and `cfgUseDILocation`) can be subsumed by checking `cfgVer`. I'm not sure if you were planning to do that in a separate PR, however.

**Review by RyanGlScott — APPROVED** [review]

File: src/Text/LLVM/PP.hs (hunk 1 above)
Comment (kquick) [reply src/Text/LLVM/PP.hs:74]: It will be in forthcoming PR's (notably in llvm-pretty-bc-parser).

File: src/Text/LLVM/PP.hs (hunk 2 above)
Comment (kquick) [reply src/Text/LLVM/PP.hs:89]: Yup, that's the plan.
